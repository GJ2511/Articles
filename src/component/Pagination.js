import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

function Pagination({ totalCount, currentPage, loading, handlePageClick }) {
    const pages = Array.from(Array(totalCount)).map((e, index) => {
        const page = index + 1;
        const active = currentPage === page;
        const classes = cx('page-item', {
            active: active,
            'hover-hand': !active,
        });

        if (loading) {
            return '';
        }
        return (
            <li className={classes} key={index} onClick={() => !active && handlePageClick(page)}>
                <p className="page-link">{page}</p>
            </li>
        );
    });

    return (
        <nav aria-label="Page navigation">
            <ul className="pagination">{pages}</ul>
        </nav>
    );
}

Pagination.propTypes = {
    currentPage: PropTypes.number.isRequired,
    handlePageClick: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired,
    totalCount: PropTypes.number.isRequired,
};

export default Pagination;
