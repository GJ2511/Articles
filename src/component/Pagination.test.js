import React from 'react';

import Pagination from './Pagination';

const props = {
    currentPage: 1,
    handlePageClick: jest.fn(),
    loading: false,
    totalCount: 10,
};

describe('<Pagination tests />', () => {
    test('match snapshot', () => {
        expect(shallow(<Pagination {...props} />)).toMatchSnapshot();
    });
});
