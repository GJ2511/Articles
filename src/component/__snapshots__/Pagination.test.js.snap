// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination tests /> match snapshot 1`] = `
<nav
  aria-label="Page navigation"
  className="table-responsive"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item active"
      key="0"
      onClick={[Function]}
    >
      <p
        className="page-link"
      >
        1
      </p>
    </li>
    <li
      className="page-item hover-hand"
      key="1"
      onClick={[Function]}
    >
      <p
        className="page-link"
      >
        2
      </p>
    </li>
    <li
      className="page-item hover-hand"
      key="2"
      onClick={[Function]}
    >
      <p
        className="page-link"
      >
        3
      </p>
    </li>
    <li
      className="page-item hover-hand"
      key="3"
      onClick={[Function]}
    >
      <p
        className="page-link"
      >
        4
      </p>
    </li>
    <li
      className="page-item hover-hand"
      key="4"
      onClick={[Function]}
    >
      <p
        className="page-link"
      >
        5
      </p>
    </li>
    <li
      className="page-item hover-hand"
      key="5"
      onClick={[Function]}
    >
      <p
        className="page-link"
      >
        6
      </p>
    </li>
    <li
      className="page-item hover-hand"
      key="6"
      onClick={[Function]}
    >
      <p
        className="page-link"
      >
        7
      </p>
    </li>
    <li
      className="page-item hover-hand"
      key="7"
      onClick={[Function]}
    >
      <p
        className="page-link"
      >
        8
      </p>
    </li>
    <li
      className="page-item hover-hand"
      key="8"
      onClick={[Function]}
    >
      <p
        className="page-link"
      >
        9
      </p>
    </li>
    <li
      className="page-item hover-hand"
      key="9"
      onClick={[Function]}
    >
      <p
        className="page-link"
      >
        10
      </p>
    </li>
  </ul>
</nav>
`;
